lib/logger.js

```diff
@@ -8,8 +8,16 @@ const LOG = cds.log('graphql')
 const { parse, visit, Kind, print } = require('graphql')
 
 const _isEmptyObject = o => Object.keys(o).length === 0
+class InvalidJSON extends Error {}
+InvalidJSON.prototype.name = 'Invalid JSON body'
+InvalidJSON.prototype.status = 400
 
-router.use(express.json({ ...cds.env.server.body_parser })) //> required by logger below
+router.use(function jsonBodyParser(req, res, next) {
+  express.json({ ...cds.env.server.body_parser })(req, res, function http_body_parser_next(err) {
+    // Need to wrap, as CAP server deliberately crashes on SyntaxErrors
+    if (err) return next(new InvalidJSON(err.message))
+  })
```

Comment:
The `next()` in the else case is still needed to continue processing the request, in case no error occurs:
```suggestion
    if (err) return next(new InvalidJSON(err.message))
    next()
  })
```

Reply:
yes yes, formatting error :)